### PR TITLE
Add data splitting script

### PR DESCRIPTION
**Pull request by jackapbutler**
* Add scripts to create data splits
* Creates 5 splits per dataset
   * train ~ (1 - 2*`val_test_splits_size`)
   * test_random ~ `val_test_splits_size` / 2 ~ composed of random papers
   * test_future ~ `val_test_splits_size` / 2 ~ composed of most recent papers
   * val_random ~ `val_test_splits_size` / 2 ~ composed of random papers
   * val_future ~ `val_test_splits_size` / 2 ~ composed of most recent papers

### CODE REVIEW

File: experiments/data/sbatch_hf_split.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#SBATCH --cpus-per-task=12
+#SBATCH --output=/fsx/proj-chemnlp/experiments/logs/split_%j.out
+#SBATCH --error=/fsx/proj-chemnlp/experiments/logs/split_%j.err
+#SBATCH --open-mode=append
+#SBATCH --account=topchem
+#SBATCH --partition=g40x
+#SBATCH --exclusive
+
+### This script runs tokenisation of Hugging Face datasets
+### The first arg ($1) is the prefix directory where the environment is saved
```
Comment (bethanyconnolly) [inline experiments/data/sbatch_hf_split.sh:14]: is $1 missing?

File: experiments/data/split_data.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "data_path", help="The full path to the original dataset.", type=str
+    )
+    parser.add_argument(
+        "val_test_splits_size",
+        help="The total test size for validation / test.",
+        type=float,
+    )
+    args = parser.parse_args()
+    name = "papers" if "papers" in args.data_path else "abstracts"
+    print("loadinggg")
```
Comment (bethanyconnolly) [inline experiments/data/split_data.py:21]: "loadinggg" :)

File: experiments/data/split_data.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    ds = datasets.load_dataset(
+        "json",
+        **{"data_files": {"train": args.data_path}, "split": "train"},
+        num_proc=os.cpu_count(),
+        keep_in_memory=True,
+    )
+
+    print("filteringgg")
```
Comment (bethanyconnolly) [inline experiments/data/split_data.py:29]: "filteringgg", is ggg a thing I don't know about? :)

File: experiments/data/split_data.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    ds = ds.filter(
+        lambda x: isinstance(x["sentences"], str) and x["pubDate"] is not None,
+        num_proc=os.cpu_count(),
+    )
+
+    print("sortinggg")
```
Comment (bethanyconnolly) [inline experiments/data/split_data.py:35]: !!!

File: experiments/data/sbatch_hf_split.sh (hunk 1 above)
Comment (jackapbutler) [reply experiments/data/sbatch_hf_split.sh:14]: oh yes, I had a short term hack without using the environment script. I've added comments and the same environment creation script now as tokenisation

**Comment by jackapbutler** [comment]
ah no the `ggg` is just random comments from me 😂 just to checkpoint where it was at when it was being very slow

**Review by bethanyconnolly — APPROVED** [review]

**Pull request merged**